packages/app-store/googlecalendar/api/callback.ts

```diff
@@ -105,23 +106,32 @@ async function getHandler(req: NextApiRequest, res: NextApiResponse) {
         appId: "google-calendar",
       },
     });
+
+    const selectedCalendarWhereUnique = {
+      userId: req.session.user.id,
+      externalId: primaryCal.id,
+      integration: "google_calendar",
+    };
+
     // Wrapping in a try/catch to reduce chance of race conditions-
     // also this improves performance for most of the happy-paths.
     try {
       await prisma.selectedCalendar.create({
         data: {
-          userId: req.session.user.id,
-          externalId: primaryCal.id,
           credentialId: credential.id,
-          integration: "google_calendar",
+          ...selectedCalendarWhereUnique,
         },
       });
     } catch (error) {
-      await prisma.credential.delete({ where: { id: credential.id } });
       let errorMessage = "something_went_wrong";
       if (error instanceof Prisma.PrismaClientKnownRequestError && error.code === "P2002") {
+        // it is possible a selectedCalendar was orphaned, in this situation-
+        // we want to recover by connecting the existing selectedCalendar to the new Credential.
+        if (await renewSelectedCalendarCredentialId(selectedCalendarWhereUnique, credential.id)) return;
+        // else
```

Comment:
I tried removing replicating this by editing credentialId to null  in 'Selected Calendar' Table. 

Now when i tried connecting the same calendar again I am stuck on this screen
probably because of this return statement.

<img width="602" alt="Screenshot 2024-02-20 at 2 53 35 PM" src="https://github.com/calcom/cal.com/assets/53316345/fca82a8d-4892-46ad-97a3-018bbeaf644b">


should it be 
```

 res.redirect(
    getSafeRedirectUrl(state?.returnTo) ??
      getInstalledAppPath({ variant: "calendar", slug: "google-calendar" })
  );

```

Reply:
Oh yeah because it doesn't return there's no return path!